OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -284,19 +285,57 @@ void onMessageActionOccurredOnPreview(@NonNull final OSInAppMessage message, @No
         action.firstClick = message.takeActionAsUnique();
 
         firePublicClickHandler(action);
+        showPrompts(message, action.prompts);
         fireClickAction(action);
+        fireTagsAndOutcomeLogs(action);
     }
 
-    private void firePrompt(OSInAppMessage message, final List<OSInAppMessagePrompt> prompts) {
-        for (OSInAppMessagePrompt prompt : prompts) {
+    private void fireTagsAndOutcomeLogs(final OSInAppMessageAction action) {
+        if (action.tags != null) {
+            OSInAppMessageTag tags = action.tags;
+            if (tags.getTagsToAdd() != null)
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Tags to add detected inside of the action click payload, ignoring because action came from IAM preview:: " + tags.getTagsToAdd());
+            if (tags.getTagsToRemove() != null)
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Tags to remove detected inside of the action click payload, ignoring because action came from IAM preview: " + tags.getTagsToRemove());
+        }
+
+        for (OSInAppMessageOutcome outcome : action.outcomes) {
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Outcome detected inside of the action click payload, ignoring because action came from IAM preview: " + outcome.toString());
+        }
+    }
```

Comment:
way too much extra code here, tags and outcomes are `JSONObjects` so just call `tags.toString()` and print that
 with a single log about the tags

Same exact thing for outcomes, `outcomes.toString()` with a log

ex.
    private void fireTagsAndOutcomeLogs(final OSInAppMessageAction action) {
 @mikechoch
mikechoch Member Pending
Don't make it specific to tags and outcomes we might have more things added here eventually

Change name from fireTagsAndOutcomeLogs to
logInAppMessagePreviewActions or
logIamPreviewActions?

```
private void logInAppMessagePreviewActions(final OSInAppMessageAction action) {
    JSONObject tags = action.tags;
    if (tags != null)
        OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Tags detected inside of the action click payload, ignoring because action came from IAM preview: " + tags.toString());
   
    JSONObject outcomes = action.outcomes;
    if (outcomes != null)
        OneSignal.onesignalLog(OneSignal.LOG_LEVEL.DEBUG, "Outcomes detected inside of the action click payload, ignoring because action came from IAM preview: " + outcomes.toString());

    // TODO: Add more action payload preview logs here in future
}
```